### ISSUE TITLE
Error when selecting some sprites: "XML Parsing Error: not well-formed"

### ISSUE THREAD

**Issue by towerofnix** [LLK/scratch-gui#1380]
### Expected Behavior

The sprite becomes selected and its blocks appear in the block workspace.

### Actual Behavior

The sprite is selected, but its blocks are nowhere to be found. The block workspace simply becomes empty.

An error is also reported in the console:

```
XML Parsing Error: not well-formed
Location: https://llk.github.io/scratch-gui/develop/#10128407
Line Number 9253, Column 39:
```

The GUI is still responsive, so I can select a different sprite, but I can't look at the blocks for the particular sprite with the error.

This is probably an issue in scratch-blocks, but I'm having trouble tracking exactly what's raising the error.

### Steps to Reproduce

Several projects have this issue, but Paper Minecraft (by griffpatch) consistently shows off the issue.

* [Open Paper Minecraft in Scratch 3.0.](https://llk.github.io/scratch-gui/develop/#10128407)
* Select either the `Steve` or `gui_invrow` sprites. Note that the blocks don't appear (and the error appears in the console).
* Note that clicking any other sprite does not have this error and the blocks appear.

### Operating System and Browser

Debian (Testing). Issue present on both Firefox 52.3.0 and Nightly (58).

**Comment by paulkaplan** [comment]
Seems to just be firefox? Cannot repro on Chrome.

**Comment by towerofnix** [comment]
>Seems to just be firefox? Cannot repro on Chrome.

@paulkaplan Yup, I figured. I noticed in another issue (LLK/scratch-vm#923) that clearly selecting Steve was working; I suppose the creator of that issue wasn't using Firefox.

**Comment by paulkaplan** [comment]
Looks like the firefox error gives a specific line number in the XML. Any chance you could dig into that xml and see what is not well-formed?

**Comment by towerofnix** [comment]
I created a simple script which checks when `parseFromString` throws an error:

```js
{
  const _parseFromString = DOMParser.prototype.parseFromString
  DOMParser.prototype.parseFromString = function(...args) {
    const ret = _parseFromString.apply(this, args)
    // see https://developer.mozilla.org/en-US/docs/Web/API/DOMParser#Error_handling
    const errElement = ret.querySelector('parsererror')
    if (errElement) {
      console.log('Error parsing; XML set on window._xml.')
      window._xml = args[0]
      console.log('Trace:')
      console.trace()
      debugger
    }
    return ret
  }
}
```

Here's a raw gist of [the badly-formed XML](https://gist.githubusercontent.com/towerofnix/419ed31c27541ce70b5976576c660260/raw/15ae34d6c259f8e4f37d52004ccb586f15e6c856/badxml.xml).

Vim shows that the bad character is a `^P` (judging by [this table](http://ee.hawaii.edu/~tep/EE160/Book/chap4/subsection2.1.1.1.html), ASCII code 16):

```xml
9249             ><value name="CONDITION"><block
9250                 id="i[3/^i-?#^r9Rd6N`f)}"
9251                 type="sensing_keypressed"
9252
9253             ><field name="KEY_OPTION">^P</field></block>
```

It looks like the issue is a "key pressed?" sensing block which is trying to check for the "^P" character... which happens to not be valid XML.

I'm unsure where this XML is being generated, but perhaps escaping special codes like that would fix the issue?

Here's [the traceback](https://gist.githubusercontent.com/towerofnix/3064884aaf431dbab4a998f0bdf21eb2/raw/9d7e90544e5640926a7e29caf479b587d79dcbd9/traceback.txt) (from `console.trace()`), if it's of any help.

**Comment by kchadha** [comment]
@towerofnix, in response to your comment about LLK/scratch-vm#923, you're right, I was using chrome. I updated that issue to add OS/browser info.

**Comment by paulkaplan** [comment]
Wow @towerofnix awesome investigation. So this is a "hacked block" where the input value was changed to include ascii char codes to extend the key pressed functionality to meta key combinations. I made a more minimal repro by remixing that project and pairing it down to a single stack: https://llk.github.io/scratch-gui/develop/#201655826

Even though it doesn't throw XML errors in chrome, it still doesn't actually render right, it seems to ignore the blocks that are inside the "IF" (here is what it should look like vs. Chrome)
![image](https://user-images.githubusercontent.com/654102/35739182-cac8212e-07fe-11e8-9156-59c03ece1f21.png)

@thisandagain @kchadha Unfortunately this is another example of hacked blocks that are going to give us trouble in 3.0 :(
